salt/states/restconf.py

```diff
@@ -0,0 +1,202 @@
+        path_check["request_restponse"] = {}
+
+    proposed_config = json.loads(
+        json.dumps(use_conf)
+    )  # convert from orderedDict to Dict (which is now ordered by default in python3.8)
+
+    log.debug("existing path config:")
+    log.debug(type(path_check["request_restponse"]))
+    log.debug(path_check["request_restponse"])
+    log.debug("proposed_config:")
+    log.debug(type(proposed_config))
+    log.debug(proposed_config)
+
+    # TODO: migrate the below == check to RecursiveDictDiffer when issue 59017 is fixed
+    if path_check["request_restponse"] == proposed_config:
+        ret["result"] = True
+        ret["comment"] = "Config is already set"
+
+    elif __opts__["test"] is True:
+        ret["result"] = None
+        ret["changes"]["changed"] = _compare_changes(
+            path_check["request_restponse"], proposed_config
+        )
+        # ret["changes"]["rest_method"] = request_method
+        ret["changes"]["rest_method_path"] = path_check["path_used"]
+        ret["comment"] = "Config will be added"
+
+    else:
+        resp = __salt__["restconf.set_data"](
+            path_check["request_path"], request_method, proposed_config
+        )
+
+        # Success
+        if resp["status"] in [201, 200, 204]:
+            ret["result"] = True
+            ret["changes"]["changed"] = _compare_changes(
+                path_check["request_restponse"], proposed_config
+            )
+            # ret["changes"]["rest_method"] = request_method
+            ret["changes"]["rest_method_path"] = path_check["path_used"]
+            ret["comment"] = "Successfully added config"
+        else:
+            ret["result"] = False
+            if "dict" in resp:
+                why = resp["dict"]
+            elif "body" in resp:
+                why = resp["body"]
+            else:
+                why = None
+            ret["comment"] = (
+                "failed to add / modify config. "
+                "API Statuscode: {}, API Response: {}, URI: {}".format(
+                    resp["status"], why, path_check["request_path"]
+                )
+            )
+            log.debug("post_content: %s", json.dumps(proposed_config))
+
+    return ret
+
+
+def _compare_changes(old, new, output_style="yaml"):
+    # option to switch to a json output
+
+    old_raw = yaml.safe_dump(old, default_flow_style=False).splitlines()
+    old = [
+        " " + x for x in old_raw
+    ]  # adding a space to start of each line to make it readable
+    new_raw = yaml.safe_dump(new, default_flow_style=False).splitlines()
+    new = [
+        " " + x for x in new_raw
+    ]  # adding a space to start of each line to make it readable
+    if output_style == "json":
+        old = json.dumps(old, sort_keys=False, indent=2).splitlines()
+        new = json.dumps(new, sort_keys=False, indent=2).splitlines()
+    log.debug("_compare_changes:")
+    log.debug("old:")
+    log.debug(old)
+    log.debug("new:")
+    log.debug(new)
```

Comment:
```suggestion
    log.debug("_compare_changes:")
    log.debug("old:")
    log.debug(old)
    log.debug("new:")
    log.debug(new)
```
Make this a single log call divided by new lines

Reply:
removed these. i don't personally think they are needed anymore.